File: src/NnCase.IR/Operators/FakeQuantize.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NnCase.IR.Operators
{
    public class FakeQuantize : Node
    {
        public InputConnector Input { get; }

        public OutputConnector Output { get; }

        public FakeQuantize(Shape inputShape)
        {
            Input = AddInput("input", DataType.Float32, inputShape);
            Output = AddOutput("output", DataType.Float32, inputShape);
        }
    }
}

File: src/ir/ops/fake_quantize.cpp
#include <ir/ops/fake_quantize.h>
#include <ir/op_utils.h>
#include <xtensor/xarray.hpp>

using namespace nncase;
using namespace nncase::ir;

fake_quantize::fake_quantize(shape_t input_shape)
{
    add_input("input", dt_float32, input_shape);
    add_output("output", dt_float32, input_shape);
}
